tests/test_generators.py

```diff
@@ -0,0 +1,72 @@
+import pandas as pd
+import pytest
+
+from transformers import pipeline
+
+from adatest import generators
+
+
+class TestTransformers:
+    @pytest.mark.parametrize("model_name", ["EleutherAI/gpt-neo-125M"])
+    def test_smoke(self, model_name):
+        hf_model = pipeline("text-generation", model=model_name)
+        target = generators.Transformers(hf_model.model, hf_model.tokenizer)
+
+        prompts = [
+            ("id A", "", "Great hotel"),
+            ("id B", "", "Bathroom too small"),
+        ]
+
+        desired_result_count = 2
+
+        results = target(
+            prompts,
+            "",
+            mode="tests",
+            topic_description="",
+            num_samples=desired_result_count,
+        )
+        assert results is not None
+        assert len(results) == desired_result_count
+        for item in results:
+            assert isinstance(item, str)
+
+
+class TestOpenAI:
+    def test_smoke(self, mocker):
+        OPENAI_API_KEY = "Not for you, CredScan"
+
+        openai_completion = mocker.patch("openai.Completion", auto_spec=True)
+        patched_response = {"choices": [{"text": "Ret 1"}, {"text": "Ret 2"}]}
```

Comment:
I like the smoke tests. Is `patched_response` supposed to match the desired output of `generators.OpenAI.__call__(...)`? If so, shouldn't this be a simple `List[str]`?

Reply:
No the patch is on the call to `openai.Completion`. I got this by stepping through a call in the debugger and capturing what OpenAI sent back.